File: backend/GestionCommerciale.Api/Models/Product.cs
namespace GestionCommerciale.Api.Models;

public class Product
{
    public int Id { get; set; }
    public string Reference { get; set; } = string.Empty;
    public string Nom { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal PrixUnitaireHT { get; set; }
    public int QuantiteEnStock { get; set; }
    public DateTime DateCreation { get; set; } = DateTime.UtcNow;

    public ICollection<OrderLine> OrderLines { get; set; } = new List<OrderLine>();
}

File: frontend/src/models/product.ts
export interface Product {
  id: number;
  reference: string;
  nom: string;
  description: string;
  prixUnitaireHT: number;
  quantiteEnStock: number;
  dateCreation: string;
}

export interface CreateProductDto {
  reference: string;
  nom: string;
  description: string;
  prixUnitaireHT: number;
  quantiteEnStock: number;
}

export interface UpdateProductDto extends CreateProductDto {}